pkg/compiler/codegen.go

```diff
@@ -900,28 +900,25 @@ func (c *codegen) Visit(node ast.Node) ast.Visitor {
 		c.currentSwitch = label
 
 		ast.Walk(c, n.X)
+		emit.Syscall(c.prog.BinWriter, "Neo.Iterator.Create")
```

Comment:
Looks intriguing. If all iterators/enumerators syscalls are to remain at cost of 1 it might even be cheaper in GAS cost!

Reply:
And they're not, in fact: `0_00000400` for most of operations and `0_01000000` for `Create`. That's quite a lot actually and it completely changes the balance. This make iterator-based approach bad for Neo 3.

Reply:
#961.